test/test.downscopedclient.ts

```diff
@@ -0,0 +1,406 @@
+                testAvailablePermission2,
+              ],
+              availabilityCondition: {
+                expression: testAvailabilityConditionExpression,
+                title: testAvailabilityConditionTitle,
+                description: testAvailabilityConditionDescription,
+              },
+            },
+          ],
+        },
+      };
+      assert.doesNotThrow(() => {
+        return new DownscopedClient(client, cabWithTwoAvailblePermissions);
+      });
+    });
+
+    it('should not throw with empty available condition', () => {
+      const cabWithNoAvailabilityCondition = {
+        accessBoundary: {
+          accessBoundaryRules: [
+            {
+              availableResource: testAvailableResource,
+              availablePermissions: [testAvailablePermission1],
+            },
+          ],
+        },
+      };
+      assert.doesNotThrow(() => {
+        return new DownscopedClient(client, cabWithNoAvailabilityCondition);
+      });
+    });
+
+    it('should not throw with only expression setup in available condition', () => {
+      const cabWithOnlyAvailabilityConditionExpression = {
+        accessBoundary: {
+          accessBoundaryRules: [
+            {
+              availableResource: testAvailableResource,
+              availablePermissions: [
+                testAvailablePermission1,
+                testAvailablePermission2,
+              ],
+              availabilityCondition: {
+                expression: testAvailabilityConditionExpression,
+              },
+            },
+          ],
+        },
+      };
+      assert.doesNotThrow(() => {
+        return new DownscopedClient(
+          client,
+          cabWithOnlyAvailabilityConditionExpression
+        );
+      });
+    });
+
+    it('should set custom RefreshOptions', () => {
+      const refreshOptions = {
+        eagerRefreshThresholdMillis: 5000,
+        forceRefreshOnFailure: true,
+      };
+      const cabWithOneAccessBoundaryRules = {
+        accessBoundary: {
+          accessBoundaryRules: [
+            {
+              availableResource: testAvailableResource,
+              availablePermissions: [testAvailablePermission1],
+              availabilityCondition: {
+                expression: testAvailabilityConditionExpression,
+              },
+            },
+          ],
+        },
+      };
+      const downscopedClient = new DownscopedClient(
+        client,
+        cabWithOneAccessBoundaryRules,
+        refreshOptions
+      );
+      assert.strictEqual(
+        downscopedClient.forceRefreshOnFailure,
+        refreshOptions.forceRefreshOnFailure
+      );
+      assert.strictEqual(
+        downscopedClient.eagerRefreshThresholdMillis,
+        refreshOptions.eagerRefreshThresholdMillis
+      );
+    });
+  });
+
+  describe('getAccessToken()', () => {
+    let sandbox: sinon.SinonSandbox;
+    before(() => {
+      const expectedAccessTokenResponse = {
+        token: 'subject_token',
```

Comment:
This is the final response. It is not the subject token. It is the access token. The subject token is exchanged for an access token.

Reply:
This is the subject token which is the access token from source credential.
And I am using it to exchange for downscoped access token

Reply:
I'll change the variable name to avoid confusion.